File: scratch_parse.py
import json

try:
    with open('C:\\\\Users\\\\ASUS\\\\en-passant-frontend-v1\\\\scratch_lighthouse.json', 'r', encoding='utf-8') as f:
        content = f.read()
        
    # Remove XML tags if present
    content = content.replace('<USER_REQUEST>', '').replace('</USER_REQUEST>', '').strip()
    
    # Try parsing
    data = json.loads(content)
    
    perf = data.get('categories', {}).get('performance', {}).get('score')
    if perf is not None:
        print(f"Performance Score: {perf * 100}")
    else:
        print("Performance score not found in JSON structure.")
except Exception as e:
    print(f"Error parsing JSON: {e}")
    # try to regex search
    import re
    match = re.search(r'"performance"\s*:\s*\{[^}]*?"score"\s*:\s*([\d\.]+)', content)
    if match:
        print(f"Regex Performance Score: {float(match.group(1)) * 100}")
    else:
        print("Regex also failed to find score.")

File: scratch_parse.js
const fs = require('fs');
let content = fs.readFileSync('C:\\\\Users\\\\ASUS\\\\en-passant-frontend-v1\\\\scratch_lighthouse.json', 'utf8');

const perfMatch = content.match(/"performance":\s*\{\s*"id":\s*"performance",\s*"title":\s*"Performance",\s*"score":\s*([\d\.]+)/);
const lcpMatch = content.match(/"largest-contentful-paint":\s*\{[^}]*?"displayValue":\s*"([^"]+)"/);
const tbtMatch = content.match(/"total-blocking-time":\s*\{[^}]*?"displayValue":\s*"([^"]+)"/);
const fcpMatch = content.match(/"first-contentful-paint":\s*\{[^}]*?"displayValue":\s*"([^"]+)"/);
const clsMatch = content.match(/"cumulative-layout-shift":\s*\{[^}]*?"displayValue":\s*"([^"]+)"/);
const siMatch = content.match(/"speed-index":\s*\{[^}]*?"displayValue":\s*"([^"]+)"/);

console.log('Performance: ' + (perfMatch ? parseFloat(perfMatch[1]) * 100 : 'not found'));
console.log('LCP: ' + (lcpMatch ? lcpMatch[1] : 'not found'));
console.log('TBT: ' + (tbtMatch ? tbtMatch[1] : 'not found'));
console.log('FCP: ' + (fcpMatch ? fcpMatch[1] : 'not found'));
console.log('CLS: ' + (clsMatch ? clsMatch[1] : 'not found'));
console.log('SI: ' + (siMatch ? siMatch[1] : 'not found'));
